ftp-native/src/main/java/org/ballerinalang/stdlib/ftp/client/FTPClient.java

```diff
@@ -99,15 +103,22 @@ public static Object initClientEndpoint(BObject clientEndpoint, BMap<Object, Obj
         return null;
     }
 
-    public static Object get(Environment env, BObject clientConnector, String filePath) throws BallerinaFTPException {
-        String url = FTPUtil.createUrl(clientConnector, filePath);
+    public static Object get(Environment env, BObject clientConnector, BString filePath, long arraySize)
+            throws BallerinaFTPException {
+        InputStream readInputStream = (InputStream) clientConnector.getNativeData(READ_INPUT_STREAM);
+        if (readInputStream != null) {
```

Comment:
If this `readInputStream` is null, then shouldn’t this `InputStream` type cast return an NPE?

Reply:
No.